Mlem/Views/Shared/Links/User/UserAvatarView.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  UserAvatarView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 05/09/2023.
+//
+
+import SwiftUI
+
+struct UserAvatarView: View {
```

Comment:
While we're refactoring, do we need this at all? Can we achieve what we need with a single `AvatarView`? It would have to take in a couple parameters to handle things like default avatar icon and clipping, but overall it would cut down on code duplication by a lot and ensure that our avatars are always styled the same way

Reply:
Sure, I can do it that way 👍